src/compiler/Uno.Compiler.Core/Syntax/UxlProcessor.cs

```diff
@@ -628,6 +629,11 @@ void WriteImageFile(ImageFile f)
                         {
                             graphics.InterpolationMode = InterpolationMode.HighQualityBicubic;
                             graphics.PixelOffsetMode = PixelOffsetMode.HighQuality;
+                            if (_env.GetString("Target").ToUpper() == "IOS")
+                            {
+                                graphics.Clear(Color.White);
+                                graphics.CompositingMode = CompositingMode.SourceOver;
+                            }
```

Comment:
So, this is better than the previous attempt, but I'm still not really convinced. As I [wrote here](https://github.com/fuse-open/uno/pull/30#issuecomment-391974467):

>>> This makes sure that if the input-image has an alpha channel, the output image will also. This leaves the "should my icon have an alpha-channel"-question up to the user, rather than a hard decision by us...

>> Apple's policy obligate us to do that, because it's not accepting the app to be uploaded to iTunesConnect even for TestFlight if the icon image has alpha channel !

> I don't think that's quite right. Our default icon does not have an alpha-channel AFAIK, and if the user has specified a custom icon with an alpha-channel, this is a user-error IMO, which can easily be fixed in the user-project. So I don't think we need to handle the case where the image has an alpha-channel other than forwarding that alpha channel.
>
> Removing the alpha channel, by blending the image on top of a background color makes assumptions about what the desired background color is, which is rather arbitrary. I'd rather not make arbitrary decisions like that.

What exactly are you trying to fix here? It sounds to me like the specified icon does indeed have translucency in it. Would it be better to fix the input-assets than to batch-process them with an arbitrary color here?

Reply:
[Here's some code](https://gist.github.com/kusma/6e554aed947b0905761491a932f61243) trying to show my point.

Now, if I run this project, I get the following output:
```
input: Format24bppRgb, output: Format32bppArgb
input: Format32bppArgb, output: Format32bppArgb
```
OK, so this is obviously not what we want, because we all of a sudden invented an alpha-channel in the first case! OK, so it might have all opaque pixels, but still. Not good.

However, if I pass `originalImg.PixelFormat` to the constructor for `Bitmap`, I get this instead:
```
input: Format24bppRgb, output: Format24bppRgb
input: Format32bppArgb, output: Format32bppArgb
```

OK, that's much better. Now the user can affect the output, and fix their projects.

I've also inspected the written files using TweakPNG, and it seems they have the format I expect them to...

Reply:
Okay let me try this, and get back.

Its good idea

Reply:
We have to agreed that there is a here 
https://github.com/fuse-open/uno/blob/master/src/compiler/Uno.Compiler.Core/Syntax/UxlProcessor.cs#L625
right?
And it still needs a contribution right ?
So while app says :
> *Keep the background simple and avoid transparency*. Make sure your icon is opaque, and don’t clutter the background. Give it a simple background so it doesn’t overpower other app icons nearby. You don’t need to fill the entire icon with content.

So the generated icon must avoid the transparency. so what is the best practice here ?

Reply:
Initially, `originalImg.PixelFormat` make things better when the input image is correct and fine and the code now preserves the situation of the image.
But should we raise an error event if the image have transparency ?

Reply:
I have full feedback:
1- If we use png image without transparency it will not work unless we add `originalImg.PixelFormat` or use `CompositingMode.SourceOver`.
2- if we use ping image with transparency it will not work at all.
3- We have to warn the use that the image contains transparency and fail the build.

Is that good? so i push a new commit ?